exam/models/exam.py

```diff
@@ -26,47 +25,63 @@ class ExtendedStudentStudent(models.Model):
                                        'Exam History', readonly=True)
 
 
-class ExtendedTimeTableLine(models.Model):
-    _inherit = 'time.table.line'
-
-    exm_date = fields.Date('Exam Date')
-    day_of_week = fields.Char('Week Day')
-
-    @api.multi
-    def on_change_date_day(self, exm_date):
-        val = {}
-        if exm_date:
-            week_day = datetime.strptime(exm_date, "%Y-%m-%d")
-            val['week_day'] = week_day.strftime("%A").lower()
-        return {'value': val}
-
-    @api.multi
-    def _check_date(self):
-        for line in self:
-            if line.exm_date:
-                dt = datetime.strptime(line.exm_date, "%Y-%m-%d")
-                if line.week_day != dt.strftime("%A").lower():
-                    return False
-                elif dt.__str__() < datetime.strptime(date.today().__str__(),
-                                                      "%Y-%m-%d").__str__():
-                    raise except_orm(_('Invalid Date Error !'),
-                                     _('Either you have selected wrong day'
-                                       'for the date or you have selected'
-                                       'invalid date.'))
-        return True
+# class ExtendedTimeTableLine(models.Model):
+#    _inherit = 'time.table.line'
+#
+#    exm_date = fields.Date('Exam Date')
+#    day_of_week = fields.Char('Week Day')
+#
+#    @api.multi
+#    def on_change_date_day(self, exm_date):
+#        val = {}
+#        if exm_date:
+#            week_day = datetime.strptime(exm_date, "%Y-%m-%d")
+#            val['week_day'] = week_day.strftime("%A").lower()
+#        return {'value': val}
+#
+#    @api.multi
+#    def _check_date(self):
+#        for line in self:
+#            if line.exm_date:
+#                dt = datetime.strptime(line.exm_date, "%Y-%m-%d")
+#                if line.week_day != dt.strftime("%A").lower():
+#                    return False
+#                elif dt.__str__() < datetime.strptime(date.today().__str__(),
+#                                                      "%Y-%m-%d").__str__():
+#                    raise except_orm(_('Invalid Date Error !'),
+#                                     _('Either you have selected wrong day'
+#                                       'for the date or you have selected'
+#                                       'invalid date.'))
+#        return True
 
 
 class ExamExam(models.Model):
     _name = 'exam.exam'
     _description = 'Exam Information'
+    
+    @api.one
+    @api.depends('exam_timetable_ids','exam_timetable_ids.standard_id')
+    def _get_standards(self):
```

Comment:
Please check with exam_timetable_ids alone, must work. if not use standard_id directly.